flow/src/main/scala/org/alephium/flow/core/TxUtils.scala

```diff
@@ -452,6 +452,244 @@ trait TxUtils { Self: FlowUtils =>
     }
   }
 
+  def partitionOutputs(
+      outputs: AVector[TxOutputInfo],
+      totalGasOpt: Option[GasBox]
+  )(implicit
+      groupConfig: GroupConfig
+  ): AVector[(AVector[TxOutputInfo], Option[GasBox])] = {
+    val outputsByGroup = outputs.groupBy(_.lockupScript.groupIndex(groupConfig))
+    val gasPerOutputOpt = totalGasOpt.map { gas =>
+      val totalDestinations = outputsByGroup.view.mapValues(_.length).values.sum
+      Math.round(gas.value.toFloat / totalDestinations) // primitive rounding
+    }
+    AVector.from(
+      outputsByGroup.values
+        .map { outputs =>
+          val gasPerGroup =
+            gasPerOutputOpt.map(gasPerOutput => GasBox.unsafe(gasPerOutput * outputs.length))
+          outputs -> gasPerGroup
+        }
+    )
+  }
+
+  def inputValueIsHigherThanOutputValue(
+      allRemainingInputs: AVector[AssetOutputInfo],
+      outputGroups: AVector[(AVector[TxOutputInfo], Option[GasBox])]
+  ): Boolean = {
+    allRemainingInputs.map(_.output.amount).iterator.foldLeft(Option(U256.Zero)) { case (acc, n) =>
```

Comment:
you can directly use [.fold](https://github.com/alephium/alephium/blob/f3c69898878a298dd87aa2b5b186a73f0dcb48e2/util/src/main/scala/org/alephium/util/AVector.scala#L396-L400) from AVector, works the same as `foldLeft`